plugins/generic/shibboleth/pages/ShibbolethHandler.inc.php

```diff
@@ -0,0 +1,437 @@
+<?php
+
+/**
+ * @file plugins/generic/shibboleth/pages/ShibbolethHandler.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2003-2017 John Willinsky
+ * Distributed under the GNU GPL v2 or later. For full terms see the file docs/COPYING.
+ *
+ * @class ShibbolethHandler
+ * @ingroup plugins_generic_shibboleth
+ *
+ * @brief Handle Shibboleth responses
+ */
+
+import('classes.handler.Handler');
+
+class ShibbolethHandler extends Handler {
+	/** @var ShibbolethAuthPlugin */
+	var $_plugin;
+
+	/** @var int */
+	var $_contextId;
+
+	/**
+	 * Intercept normal login/registration requests; defer to Shibboleth.
+	 * 
+	 * @param $args array
+	 * @param $request Request
+	 * @return bool
+	 */
+	function activateUser($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function changePassword($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function index($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function lostPassword($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function register($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function registerUser($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function requestResetPassword($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * @copydoc ShibbolethHandler::activateUser()
+	 */
+	function savePassword($args, $request) {
+		return $this->_shibbolethRedirect($args, $request);
+	}
+
+	/**
+	 * Login handler; receives post-validation Shibboleth redirect.
+	 * 
+	 * @param $args array
+	 * @param $request Request
+	 * @return bool
+	 */
+	function shibLogin($args, $request) {
+		$this->_plugin = $this->_getPlugin();
+		$this->_contextId = $this->_plugin->getCurrentContextId();
+		$uinHeader = $this->_plugin->getSetting(
+			$this->_contextId,
+			'shibbolethHeaderUin'
+		);
+		$emailHeader = $this->_plugin->getSetting(
+			$this->_contextId,
+			'shibbolethHeaderEmail'
+		);
+
+		// We rely on these headers being present.
+		if (!isset($_SERVER[$uinHeader])) {
+			syslog(
```

Comment:
@crism, we've generally used `error_log` rather than `syslog` in the past -- is there a reason for introducing a different logging pattern here? I'd worry that admins won't be used to looking there for OJS-related stuff.

Reply:
The previous Shibboleth plugin used `syslog()` and I tried to reuse as much of that code as possible. I’d rather use `error_log()`.